lib/ruby_smb/client/negotiation.rb

```diff
@@ -57,15 +57,19 @@ def negotiate_request
 
       # Takes the raw response data from the server and tries
       # parse it into a valid Response packet object.
-      # This method currently assumes that all SMB1 will use Extended Security.
       #
       # @param raw_data [String] the raw binary response from the server
       # @return [RubySMB::SMB1::Packet::NegotiateResponseExtended] when the response is an SMB1 Extended Security Negotiate Response Packet
       # @return [RubySMB::SMB2::Packet::NegotiateResponse] when the response is an SMB2 Negotiate Response Packet
       def negotiate_response(raw_data)
         response = nil
         if smb1
-          packet = RubySMB::SMB1::Packet::NegotiateResponseExtended.read raw_data
+          packet = RubySMB::SMB1::Packet::NegotiateResponse.read raw_data
+
+          if packet.smb_header.flags2.extended_security == 1
+            packet = RubySMB::SMB1::Packet::NegotiateResponseExtended.read raw_data
+          end
+
           response = packet if packet.valid?
```

Comment:
Apparently,  the [extended security capability](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb/d367854f-5eee-45e8-a588-eed596a1a521) flag needs to be checked in the server response instead of the SMB header flag2 `extended_security` flag. This is what the [`valid?`](https://github.com/rapid7/ruby_smb/blob/master/lib/ruby_smb/smb1/packet/negotiate_response_extended.rb#L41) method do.

I also believe it is more likely to receive an `NegotiateResponseExtended` response than a `NegotiateResponse` response. So, an improvement would be to parse with the `NegotiateResponseExtended` structure first and call `valid?`, which is what the original code do. Then, if it is not valid, parse with the `NegotiateResponse` structure. This logic can be added to the `if response.nil?` block below.

Reply:
Cheers man 👍